assets/js/base/components/payment-methods/style.scss

```diff
@@ -31,15 +32,14 @@ $border-width: 1px;
 			border-radius: 0 5px 0 0;
 			content: "";
 			display: block;
+			margin-left: $gap-small;
 			pointer-events: none;
 			flex-grow: 1;
 		}
 	}
 
 	.wc-block-components-express-checkout__title {
 		flex-grow: 0;
-		padding-left: $gap-small;
-		padding-right: $gap-small;
```

Comment:
I can verify this change fixes the gap on the express checkout title:

<img width="700" alt="Image 2020-06-12 at 3 42 43 PM" src="https://user-images.githubusercontent.com/1429108/84540475-6cab7e80-acc3-11ea-92a5-b970c545aa83.png">

Reply:
We had two selectors with the same specificity competing. The issue happened because depending on the build mode, stylesheets were loaded in a different order, so the styles applied where different.

> If this is a style loading issue, should we document this in an issue for further investigation?

Created an issue: #2713.